internal/auth0/user/resource_permission_test.go

```diff
@@ -0,0 +1,117 @@
+package user_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+
+	"github.com/auth0/terraform-provider-auth0/internal/acctest"
+)
+
+const givenAResourceServerWithPermissions = `
+resource "auth0_resource_server" "resource_server" {
+	name = "Acceptance Test - {{.testName}}"
+	identifier = "https://uat.api.terraform-provider-auth0.com/{{.testName}}"
+	scopes {
+		value = "read:foo"
+		description = "Can read Foo"
+	}
+	scopes {
+		value = "create:foo"
+		description = "Can create Foo"
+	}
+}
+`
+
+const givenAUserPermission = `
+resource "auth0_user_permission" "user_permission_read" {
+	depends_on = [ auth0_resource_server.resource_server, auth0_user.user ]
+
+	user_id = auth0_user.user.id
+	resource_server_identifier = auth0_resource_server.resource_server.identifier
+	permission = "read:foo"
+}
+`
+
+const givenAnotherUserPermission = `
+resource "auth0_user_permission" "user_permission_create" {
+	depends_on = [ auth0_resource_server.resource_server, auth0_user.user ]
+
+	user_id = auth0_user.user.id
+	resource_server_identifier = auth0_resource_server.resource_server.identifier
+	permission = "create:foo"
+}
+`
+
+const testAccUserPermissionNoneAssigned = givenAResourceServerWithPermissions + testAccUserEmpty
```

Comment:
Because there's no `depends_on` within the `testAccUserEmpty` config, when running the `testAccUserPermissionNoneAssigned` scenario, sometimes the API call that runs first is the one to create the resource server and at other times it's the user, that makes our tests sometimes fail because the execution order of saved HTTP requests is not consistent. 

I would suggest having a 

```go
const givenAResourceServerWithPermissionsAndAUser = `
resource "auth0_resource_server" "resource_server" {
	name = "Acceptance Test - {{.testName}}"
	identifier = "https://uat.api.terraform-provider-auth0.com/{{.testName}}"
	scopes {
		value = "read:foo"
		description = "Can read Foo"
	}
	scopes {
		value = "create:foo"
		description = "Can create Foo"
	}
}

resource auth0_user user {
	depends_on = [ auth0_resource_server.resource_server ]

	connection_name = "Username-Password-Authentication"
	user_id = "{{.testName}}"
	username = "{{.testName}}"
	password = "passpass$12$12"
	email = "{{.testName}}@acceptance.test.com"
}
`
```